src/main/java/org/alliancegenome/curation_api/services/ReferenceService.java

```diff
@@ -45,6 +51,23 @@ public ObjectResponse<Reference> getByCurie(String curie) {
 	@Transactional
 	public Reference retrieveFromDbOrLiteratureService(String curieOrXref) {
 		Reference reference = null;
+		if (referenceRequest != null) {
+			if (referenceCacheMap.containsKey(curieOrXref)) {
+				reference = referenceCacheMap.get(curieOrXref);
+			} else {
+				Log.debug("Reference not cached, caching reference: (" + curieOrXref + ")");
+				reference = findOrCreateReference(curieOrXref);
+				referenceCacheMap.put(curieOrXref, reference);
+			}
+		} else {
+			reference = findOrCreateReference(curieOrXref);
+			referenceRequest = new Date();
```

Comment:
I would think you would want to put it in the map here also.

Reply:
PS not sure what the date is doing here... typically the logic goes:
if(exists in map) return ref;
else { lookup and cache in map) return ref;

Reply:
Same logic we’ve used on all other caches we’ve implemented. Doesn’t get cached until 2nd time it’s seen so that only common (well, more than 1 at least) objects get stored.  Date could just as well be a Boolean I suppose.